frontend/ui/widget/menu.lua

```diff
@@ -271,12 +272,24 @@ function MenuItem:init()
                 fgcolor = self.dim and Blitbuffer.COLOR_GREY or nil,
             }
             local height = item_name:getSize().h
-            if height < self.dimen.h - 2 * self.linesize then -- we fit !
+            if height < max_item_height or flag_fit then -- we fit !
                 break
             end
-            -- Don't go too low, and then truncate text
-            if self.font_size < 12 then
-                self.text = self.text:sub(1, -5) .. "…"
+            -- Don't go too low, and then decrease lines
+            if self.font_size <= 12 then
+                local line_height = height / #item_name.vertical_string_list -- should be an integer
+                local lines = math.floor(max_item_height / line_height)
+                local offset = item_name.vertical_string_list[lines + 1].offset - 2
```

Comment:
I don' t think it needs a `-2`, not even the `-1` i thought. It looks to me like it work best without any substraction.
So, may be here, to avoid the thing I got once but can't reproduce:
```lua
                local offset
                if item_name.vertical_string_list[lines + 1] then
                    offset = item_name.vertical_string_list[lines + 1].offset
                else -- shouldn't happen, but just in case
                    offset = #item_name.char_width_list
                end
```

Reply:
Thanks, I made changes according to your suggestions.